ozaria/site/components/cutscene/PageCutscene/index.vue

```diff
@@ -0,0 +1,81 @@
+<script>
+import LayoutChrome from '../../common/LayoutChrome'
+import LayoutCenterContent from '../../common/LayoutCenterContent'
+import BaseVideo from '../common/BaseVideo'
+
+const CUTSCENE_ASPECT_RATIO = 16 / 9
+
+module.exports = Vue.extend({
+  props: {
+    cutsceneId: {
+      type: String,
+      required: true
+    }
+  },
+  data: () => ({
+    width: 1920,
+    height: 1080,
+  }),
+  components: {
+    'layout-chrome': LayoutChrome,
+    'layout-center-content': LayoutCenterContent,
+    'base-video': BaseVideo
+  },
+  mounted: function() {
+    if (!me.hasCutsceneAccess()) {
+      return application.router.navigate('/', { trigger: true })
+    }
+
+    window.addEventListener('resize', this.onResize)
+    this.onResize()
+
+  },
+  destroyed() {
+    window.removeEventListener("resize", this.onResize)
+  },
+  methods: {
+    onResize: function() {
+      const userWidth = window.innerWidth
+        || document.documentElement.clientWidth
+        || document.body.clientWidth
+
+      const userHeight = window.innerHeight
+        || document.documentElement.clientHeight
+        || document.body.clientHeight
+
+      const height = this.height = Math.min(userWidth / CUTSCENE_ASPECT_RATIO, userHeight)
+      const width = this.width = Math.min(userWidth, userHeight * CUTSCENE_ASPECT_RATIO)
+    }
+  }
+})
+</script>
+
+<template>
+  <layout-chrome>
+    <layout-center-content>
+      <div id="cutscene-player"
+      :style="{ width: width+'px', height: height+'px' }">
+        <base-video
+    videoSrc="https://assets.koudashijie.com/CoCo%E7%AE%80%E4%BB%8B.mp4"
+    :captions="[{
+      label:'English captions',
+      src:'/captions/example.vtt',
+      srclang:'en'
```

Comment:
Is the video source and captions hard-coded for testing purposes? Can we add a comment here describing that?
Also, will there be a cutscenes collection in the database to store this information, and fetch it based on cutscene id/slug?

Reply:
Yes. I can add a comment. This is currently hard coded, but will be fetched via a slug from the `cutscene` collection.